src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -110,15 +110,15 @@ public function transform($entityOrCollection)
 
         if ($this->multiple) {
             $isArray = \is_array($entityOrCollection);
-            if (!$isArray && substr(\get_class($entityOrCollection), -1 * \strlen($this->className)) === $this->className) {
+            if (!$isArray && substr(\get_class($entityOrCollection), -1 * \strlen((string)$this->className)) === $this->className) {
```

Comment:
According to the property and constructor type-hint, `$this->className` is already a `string`. If it's something else it means it is probably your code that needs fixing.

Alternatively maybe we could add type declarations to the constructor to ensure the correct type is passed to the class.

Reply:
I'm sorry for my ignorance, the person responsible for the project is sick, but **this happens when calling an edit from an Admin**, and we're following the documentation of [ModelAutocompleteType](https://symfony.com/doc/master/bundles/SonataAdminBundle/reference/form_types.html#sonataadminbundleformtypemodelautocompletetype) is correct (By the dump it is what is calling `ModelToIdPropertyTransformer`). They further define that the class field should be null.

> class
The entity class managed by this field. Defaults to null, but is actually calculated from the linked admin class. You usually should not need to set this manually.

What I'm missing from the docs?

Reply:
> but is actually calculated from the linked admin class

that's the important part, it should never be null because the admin is filling this field instead. Maybe something is happening that prevent the admin from guessing the target class.